stable/elasticsearch-exporter/templates/service.yaml

```diff
@@ -19,6 +19,10 @@ spec:
   ports:
     - name: {{ .Values.service.metricsPort.name }}
       port: {{ .Values.service.httpPort }}
+      {{- if eq .Values.service.type "NodePort"}}
```

Comment:
nodePort can be valid for LoadBalancer type too. Maybe it is better not to have too much intelligence here and just have two conditions (if targetPort, then set it, if nodePort, then set it).